repos/keg-components/src/hooks/useKeyPress.js

```diff
@@ -0,0 +1,47 @@
+import { useState, useCallback, useEffect } from 'react'
+import { getPlatform } from 'KegGetPlatform'
+import { noOp } from '@keg-hub/jsutils'
+
+const platform = getPlatform()
+
+/**
+ * Helper for `useKeyPress` that sets and removes window listeners
+ * @param {Function} upHandler 
+ * @param {Function} downHandler 
+ */
+const manageListeners = (upHandler, downHandler) => {
+  window.addEventListener('keydown', downHandler)
+  window.addEventListener('keyup', upHandler)
+
+  // cleanup
+  return () => {
+    window.removeEventListener('keydown', downHandler)
+    window.removeEventListener('keyup', upHandler)
+  }
+}
+
+/**
+ * @param {string} targetKey
+ * @return {boolean} whether the key was pressed down or not
+ */
+export const useKeyPress = targetKey => {
```

Comment:
I get why the hooks are before the final `useEffect` hook;
but I think it would be safe to move the `platform === 'web'` check up to the top of this method and here's why
* React requires the same amount of hooks to be called on every render relative to the components that are rendered
  * Which I'm pretty sure is the reason you put the `useCallback` hooks here   
* But, on native, we never want any of these hooks to run
* So if we more the `platform === 'web'` to the top of this hook, it should never call them on native.
  * Because they are never called, React doesn't care about them, and we should never have an issue if hooks being conditionally called

Personally, I find it easier to just separate the native and web hook versions into their own file
* This way you just let the resolver load the correct file, and you don't have to add extra logic into the hook to manage the platform
* It ends up being less code in the final bundle, and easier to reason about.
* It's just generally cleaner

I'm open to your thoughts on this, but I would prefer our other hooks such as `useClassList`, `useScroll`, and `useAnimate`

Reply:
Totally agree. Updated to be consistent with the others.